unit-tests/src/test/scala/java/lang/ThrowablesSuite.scala

```diff
@@ -0,0 +1,356 @@
+           s"message: '${resultMessage}' != expected: '${expectedMessage}'")
+  }
+
+  private def checkStackTraceString(trace: String): Unit = {
+    val startText = "java.lang.Throwable"
+    assert(trace.startsWith(startText),
+           s"Expected trace to start with '${startText}' and it did not.")
+
+    val containsText = "\tat <none>.main(Unknown Source)"
+    assert(trace.contains(containsText),
+           s"Expected trace to contain '${containsText}' and it did not.")
+  }
+
+  private def checkStackTrace(throwable: Throwable): Unit = {
+    val sw = new java.io.StringWriter
+    val pw = new java.io.PrintWriter(sw)
+
+    throwable.printStackTrace(pw)
+
+    checkStackTraceString(sw.toString)
+  }
+
+  private def checkSuppressed(throwable: Throwable, expectedLength: Int) {
+    val getSuppressedLength = throwable.getSuppressed.length
+    assert(getSuppressedLength == expectedLength,
+           s"getSuppressed.length: ${getSuppressedLength} != " +
+             s"expected: ${expectedLength}")
+  }
+
+  private def checkConstructed(throwable: Throwable,
+                               expectedMessage: String,
+                               expectedCause: Throwable,
+                               expectedSuppressedLength: Int): Unit = {
+
+    checkMessage(throwable, expectedMessage)
+    checkCause(throwable, expectedCause)
+    checkSuppressed(throwable, 0)
+    checkStackTrace(throwable)
+  }
+
+  // Zero & two argument constructor tests will exercise cases where
+  // the rightmost two arguments are true.
+
+  test("Throwable(message, cause, false, false)") {
+    val expectedMessage = "Athchomar chomakea"
+    val expectedCause   = new Throwable("Khal Drogo")
+
+    val throwable = new Throwable(expectedMessage, expectedCause, false, false)
+
+    checkConstructed(throwable, expectedMessage, expectedCause, 0)
+
+  }
+
+  test("Throwable(message)") {
+    val expectedMessage = "Hello World"
+    val expectedCause   = null
+
+    val throwable = new Throwable(expectedMessage)
+
+    checkConstructed(throwable, expectedMessage, expectedCause, 0)
+
+  }
+
+  test("Throwable(cause)") {
+    val expectedMessageStem = "Primum Mobile"
+    val expectedMessage     = s"java.lang.Throwable: ${expectedMessageStem}"
+
+    val expectedCause = new Throwable(expectedMessageStem)
+
+    val throwable = new Throwable(expectedCause)
+
+    checkConstructed(throwable, expectedMessage, expectedCause, 0)
+
+  }
+
+  test("Throwable()") {
+    val expectedMessage = null
+    val expectedCause   = null
+
+    val throwable = new Throwable()
+
+    checkConstructed(throwable, expectedMessage, expectedCause, 0)
+
+  }
+
+  // Thirteen public methods are documented for class Throwable.
+  //
+  // Five (5) methods: fillinStackTrace(), getCause(), getMessage(),
+  // getSuppressed(), printStackTrace(PrintWriter), have been exercised by
+  // constructor tests above and do not need separate tests. This is the
+  // Thetis problem.
```

Comment:
The "Thetis problem" does not ring any bell to me as a computer science thing. Perhaps this sentence can be removed?

Reply:
Removed. Thank you.